addons/website_sale/controllers/main.py

```diff
@@ -1125,30 +1125,37 @@ def add_product(self, name=None, category=None, **post):
         })
         return "%s?enable_editor=1" % product.product_tmpl_id.website_url
 
-    @http.route(['/shop/change_sequence'], type='json', auth='user')
-    def change_sequence(self, id, sequence):
-        product_tmpl = request.env['product.template'].browse(id)
-        if sequence == "top":
-            product_tmpl.set_sequence_top()
-        elif sequence == "bottom":
-            product_tmpl.set_sequence_bottom()
-        elif sequence == "up":
-            product_tmpl.set_sequence_up()
-        elif sequence == "down":
-            product_tmpl.set_sequence_down()
-
-    @http.route(['/shop/change_size'], type='json', auth='user')
-    def change_size(self, id, x, y):
-        product = request.env['product.template'].browse(id)
-        return product.write({'website_size_x': x, 'website_size_y': y})
-
-    @http.route(['/shop/change_ppg'], type='json', auth='user')
-    def change_ppg(self, ppg):
-        request.env['website'].get_current_website().shop_ppg = ppg
-
-    @http.route(['/shop/change_ppr'], type='json', auth='user')
-    def change_ppr(self, ppr):
-        request.env['website'].get_current_website().shop_ppr = ppr
+    @http.route(['/shop/config/product'], type='json', auth='user')
+    def change_product_config(self, product_id, **options):
+        if not request.env.user.has_group('website.group_website_publisher'):
+            raise NotFound()
+
+        product = request.env['product.template'].browse(product_id)
+        if "sequence" in options:
+            sequence = options["sequence"]
+            if sequence == "top":
+                product.set_sequence_top()
+            elif sequence == "bottom":
+                product.set_sequence_bottom()
+            elif sequence == "up":
+                product.set_sequence_up()
+            elif sequence == "down":
+                product.set_sequence_down()
+        if {"x", "y"} <= set(options):
+            product.write({'website_size_x': options["x"], 'website_size_y': options["y"]})
+
+    @http.route(['/shop/config/website'], type='json', auth='user')
+    def _change_website_config(self, **options):
+        if not request.env.user.has_group('website.group_website_publisher'):
+            raise NotFound()
+
+        current_website = request.env['website'].get_current_website()
+        if "ppg" in options:
+            current_website.shop_ppg = options["ppg"]
```

Comment:
```
if options.get('ppg'):
   current_website.shop_ppg = options["ppg"] or 1
```

Avoid wrong config where we receive '' or or 0

what you think about it ?

Reply:
Makes sense, never too many sanity checks